class-gf-razorpay.php

```diff
@@ -0,0 +1,319 @@
+                  'meta_name' => 'billingInformation_email'
+            ),
+            array('name'      => 'address1',
+                  'label'     => 'Address',
+                  'meta_name' => 'billingInformation_address'
+            ),
+            array('name'      => 'address2',
+                  'label'     => 'Address 2',
+                  'meta_name' => 'billingInformation_address2'
+            ),
+            array('name'      => 'city',
+                  'label'     => 'City',
+                  'meta_name' => 'billingInformation_city'
+            ),
+            array('name'      => 'state',
+                  'label'     => 'State',
+                  'meta_name' => 'billingInformation_state'
+            ),
+            array('name'      => 'zip',
+                  'label'     => 'Zip',
+                  'meta_name' => 'billingInformation_zip'
+            ),
+            array('name'      => 'country',
+                  'label'     => 'Country',
+                 'meta_name'  => 'billingInformation_country'
+            ),
+        );
+    }
+
+    public function callback()
+    {
+        $entryId = $_COOKIE[self::RAZORPAY_ENTRY_ID];
+
+        $razorpayOrderId = $_COOKIE[self::RAZORPAY_ORDER_ID];
+
+        $entry = GFAPI::get_entry($entryId);
+
+        $api = new Api($this->get_plugin_setting(self::GF_RAZORPAY_KEY),
+            $this->get_plugin_setting(self::GF_RAZORPAY_SECRET));
+
+        $attributes = array (
+            'razorpay_order_id'   => $razorpayOrderId,
+            'razorpay_payment_id' => rgpost('razorpay_payment_id'),
+            'razorpay_signature'  => rgpost('razorpay_signature'),
+        );
+
+        $success = false;
+
+        if (($entry !== null) and
+            (empty(rgpost('razorpay_payment_id')) === false) and
+            (empty(rgpost('razorpay_signature')) ===false))
+        {
+            try
+            {
+                $api->utility->verifyPaymentSignature($attributes);
+                $success = true;
+            }
+            catch (Errors\SignatureVerificationError $e)
+            {
+                $success = false;
+            }
+        }
+
+        $action = array(
+            'id'             => $attributes['razorpay_payment_id'],
+            'type'           => 'complete_payment',
+            'transaction_id' => $attributes['razorpay_payment_id'],
+            'amount'         => $entry['payment_amount'],
+            'entry_id'       => $entry['id']
+        );
+
+        if ($success === false)
+        {
+            $action['type'] = 'fail_payment';
+        }
+
+        return $action;
+    }
+
+    public function post_callback($callback_action, $callback_result)
+    {
+        $entry               = GFAPI::get_entry($callback_action['entry_id']);
+        $feed                = $this->get_payment_feed($entry );
+
+        do_action('gform_razorpay_post_payment_' . $callback_action['type'],  $_POST, $entry, $feed);
```

Comment:
@riagarg5 Could you please use
`
"gform_razorpay_post_payment_$callback_action['type']"
`